File: api/src/test/java/net/blueshell/api/factory/dto/event/EventFeedbackDTOFactory.java
package net.blueshell.api.factory.dto.event;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.event.EventFeedbackDTO;
import net.blueshell.api.factory.dto.BaseDtoFactory;
import org.springframework.stereotype.Component;

/**
 * Factory for EventFeedbackDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class EventFeedbackDTOFactory extends BaseDtoFactory<EventFeedbackDTO> {

    @Override
    public Class<EventFeedbackDTO> targetType() {
        return EventFeedbackDTO.class;
    }

    @Override
    public EventFeedbackDTO createBasic() {
        EventFeedbackDTO dto = new EventFeedbackDTO();
        dto.setFeedback("Great event!");
        dto.setEventId(nextId());
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/event/EventFeedbackDTOFactory.kt
package net.blueshell.api.factory.dto.event

import net.blueshell.api.dto.event.EventFeedbackDTO
import net.blueshell.api.factory.dto.BaseDtoFactory
import org.springframework.stereotype.Component

/**
 * Factory for EventFeedbackDTO test instances.
 */
@Component
class EventFeedbackDTOFactory : BaseDtoFactory<EventFeedbackDTO>() {

    override fun targetType(): Class<EventFeedbackDTO> = EventFeedbackDTO::class.java

    override fun createBasic(): EventFeedbackDTO {
        val dto = EventFeedbackDTO()
        dto.feedback = "Great event!"
        dto.eventId = nextId()
        return dto
    }
}
